analytics.js

```diff
@@ -1337,10 +1337,30 @@ const eventsPushedAlready =
   window.rudderanalytics.push == Array.prototype.push;
 
 const argumentsArray = window.rudderanalytics;
-
-while (argumentsArray && argumentsArray[0] && argumentsArray[0][0] !== "load") {
-  argumentsArray.shift();
+/**
+ * Iterate the call stack until we find load call and
+ * then shift it to the beginning.
+ *
+ * Ex: Say the call stack currently have [page, identify, load, track]
+ * It will become [load, page, identify, track]
+ */
+let i = 0;
+let loadPresent = false;
+while (i < argumentsArray.length) {
+  if (argumentsArray[i] && argumentsArray[i][0] === "load") {
+    argumentsArray.unshift(argumentsArray.splice(i, 1)[0]);
+    loadPresent = true;
+    break;
+  }
+  i += 1;
 }
+/**
+ * When load is not present in the call stack restrict other events from proceeding
+ */
+if (!loadPresent) {
+  argumentsArray.length = 0;
+}
```

Comment:
There is no need for this check. Instead, you can add an else statement on Ln 1374.

Reply:
Addressed